src/SAWScript/Panic.hs

```diff
@@ -0,0 +1,34 @@
+{- |
+Module      : SAWScript.Panic
+License     : BSD3
+Maintainer  : huffman
+Stability   : provisional
+-}
+
+{-# LANGUAGE Trustworthy, TemplateHaskell #-}
+module SAWScript.Panic
+  (HasCallStack, panic) where
+
+import Panic hiding (panic)
+import qualified Panic as Panic
+
+data Saw = Saw
+
+-- | Raise a fatal error. The purpose of 'panic' is to indicate
+-- conditions caused by programmer errors (e.g. violation of datatype
+-- invariants), not problems caused by bad user input.
+panic ::
+  HasCallStack =>
+  -- | Location of problem
+  String ->
+  -- | Problem description (lines)
+  [String] ->
+  a
+panic = Panic.panic Saw
+
+instance PanicComponent Saw where
+  panicComponentName _ = "Saw"
```

Comment:
This should probably be "SAW" since that's the user-visible name (and to avoid confusion).  I don't see it in your sample output, but I'm assuming this is displayed somewhere.

Reply:
I was kind of thinking the same thing.